src/custom-providers/azext_custom_providers/custom.py

```diff
@@ -0,0 +1,52 @@
+# --------------------------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See License.txt in the project root for license information.
+# --------------------------------------------------------------------------------------------
+# pylint: disable=line-too-long
+# pylint: disable=too-many-statements
+# pylint: disable=too-many-lines
+# pylint: disable=too-many-locals
+# pylint: disable=unused-argument
+from azure.cli.core.util import sdk_no_wait
+
+
+def create_custom_providers_custom_resource_provider(client,
+                                                     resource_group_name,
+                                                     resource_provider_name,
+                                                     location=None,
+                                                     tags=None,
+                                                     actions=None,
+                                                     resource_types=None,
+                                                     validations=None,
+                                                     no_wait=False):
+    body = {'location': location, 'tags': tags,
+            'actions': actions, 'resource_types': resource_types,
+            'validations': validations}
+    return sdk_no_wait(no_wait, client.create_or_update, resource_group_name=resource_group_name, resource_provider_name=resource_provider_name, resource_provider=body)
+
+
+def update_custom_providers_custom_resource_provider(client,
```

Comment:
can we update other fileds?

Reply:
No, only support tags

---
In reply to: [422454624](https://github.com/Azure/azure-cli-extensions/pull/1627#discussion_r422454624) [](ancestors = 422454624)